src/mkMarkdown.ts

```diff
@@ -0,0 +1,25 @@
+import * as vscode from 'vscode';
+import { splitIfPanelExists } from './extension';
+
+export function mkMarkdown(): (...args: any[]) => any {
+  return async () => {
+    const currentPanel = vscode.window.activeTextEditor ? vscode.window.activeTextEditor.viewColumn : undefined;
+    const uri = vscode.Uri.parse('markdown:' + 'L4'); // 'name of tab itself 
+    const doc = await vscode.workspace.openTextDocument(uri); // calls back into the provider
+
+    // ensure new panel opens instead of new tab
+    splitIfPanelExists(currentPanel);
+
+    await vscode.window.showTextDocument(doc, vscode.ViewColumn.Beside);
+  };
+}
+
+ // Register virtual doc provider
+ export const myScheme = 'markdown';
```

Comment:
Do we have to use `markdown` scheme here?

Reply:
registerTextDocumentContentProvider in extension.ts expects 2 arguments so there has to be a scheme; I've used 'markdown' as a placeholder but it could be any string (since we're not using a module like cowsay in the tutorial). Since the scheme is just a string though, I'll plug it into registerTextDocumentContentProvider directly instead of exporting it.